### ISSUE TITLE
Test live-reload functionality on different platforms and technologies

### ISSUE THREAD

**Issue by joheriks** [vaadin/flow#7836]
## Scenario
* Import skeleton starter (plain/spring for Spring-devtools) to IDE, use version 16.0-SNAPSHOT
* Set up the live reload technology as required for the IDE
* Start app
* Make a code change and recompile
* Verify that browser automatically reloads

## Test matrix
|  | IntelliJ Windows  | IntelliJ Linux | IntelliJ macOS |  Eclipse Windows  | Eclipse Linux | Eclipse macOS | VSCode | NetBeans |
| - | - | - | - | - | - | - | - | - |
| Spring DevTools | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ (W10) | ✅|
| HotswapAgent | ✅ | ✅ | ✅ |  ✅ | ✅ | ✅ |  ✅ ❌ | ✅| 
| JRebel | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | 🙅‍♂ | ✅† |

† Tested by running on Tomcat in NetBeans instead of Maven `jetty:run`

## Installing and using HotswapAgent
*  Install DCEVM JDK 8 (http://dcevm.github.io/), alternatively use JDK 11 with integrated HotswapAgent (https://github.com/TravaOpenJDK/trava-jdk-11-dcevm/releases) and disable it by passing `-XX:+DisableHotswapAgent` before `-javaagent:...`
* Download snapshot version from https://github.com/HotswapProjects/HotswapAgent
* `mvn install`
* Run with `-javaagent:<userdir>/.m2/repository/org/hotswapagent/hotswap-agent/1.4.1-SNAPSHOT/hotswap-agent-1.4.1-SNAPSHOT.jar`

## JRebel
* Add module `flow-jrebel-plugin` to the project
* Install JRebel plugin into IDE / use as Java agent if no plugin available

**Comment by joheriks** [comment]
Note on JRebel: While live reload works, during server starup (and when running on server, at each redeploy) the following error is logged when using Vaadin 14.2 or newer:
```
2020-03-18 22:17:56 JRebel: ERROR An error occured in a request listener while handling raw request: java.lang.NoSuchMethodError: 'com.vaadin.flow.server.startup.ApplicationRouteRegistry com.vaadin.flow.server.startup.ApplicationRouteRegistry.getInstance(javax.servlet.ServletContext)'
```

**Comment by jm-ferreira** [comment]
LiveReload works using HotswapAgent. Although, there's an additional flag that needs to be turned on in addition to the steps in addition to what's documented in [http://hotswapagent.org/mydoc_setup_netbeans.html](the Hotswap Docs).

When setting up the debugger enable autoswap using `=autoHotswap=true`. Set properties should look like this:
`Env.MAVEN_OPTS=-XXaltjvm=dcevm -javaagent:<userdir>/.m2/repository/org/hotswapagent/hotswap-agent/1.4.1-SNAPSHOT/hotswap-agent-1.4.1-SNAPSHOT.jar=autoHotswap=true`

**Comment by pleku** [comment]
For VSCode & TravaOpenJDK and project base with Jetty:

Using the older version of the hotswapagent, it worked but there were some errors in console and some extra webpack compilations due to forgetting `<scanIntervalSeconds>`. It might be that it was not working properly and what I was seeing was due to jetty restart. 🤦‍♂ 

After building manually a new `hotswapagent.jar` and disabling `scanIntervalSecods` I could get the automatic reload to work - but the UI did not reflect to the code changes in a class. Also as I ran Jetty using the plugin window in VSCode, it was quite unstable and crashed after stopping the Jetty process.

I'd say that there is some instability with using VSCode, Jetty and Hotswapagent and we should not recommend that combo as there is something going wrong. I would not spend time now investigating this as there are other working alternatives.

In case someone wants to retest with VSCode, remember to set the automatic Java compilation on and also enable the hotswap implicitly with a property file or jvm param https://github.com/HotswapProjects/HotswapAgent/blob/master/hotswap-agent-core/src/main/resources/hotswap-agent.properties#L44
